java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/OsmGpxController.java

```diff
@@ -0,0 +1,202 @@
+package net.osmand.server.controllers.pub;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import net.osmand.data.LatLon;
+import net.osmand.server.DatasourceConfiguration;
+import net.osmand.shared.gpx.GpxFile;
+import net.osmand.shared.gpx.GpxTrackAnalysis;
+import net.osmand.shared.gpx.GpxUtilities;
+import net.osmand.shared.gpx.primitives.WptPt;
+import net.osmand.util.Algorithms;
+import okio.Buffer;
+import okio.Source;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.geojson.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.web.bind.annotation.*;
+
+import java.io.IOException;
+import java.sql.Array;
+import java.sql.ResultSet;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@RestController
+@RequestMapping("/osmgpx")
+public class OsmGpxController {
+
+	@Autowired
+	@Qualifier("osmgpxJdbcTemplate")
+	JdbcTemplate jdbcTemplate;
+
+	@Autowired
+	DatasourceConfiguration config;
+
+	@Autowired
+	Gson gson = new GsonBuilder().create();
+
+	protected static final Log LOGGER = LogFactory.getLog(OsmGpxController.class);
+
+	Map<String, RouteFile> routesCache = new ConcurrentHashMap<>();
+
+	private static final int MAX_RUNTIME_CACHE_SIZE = 5000;
+	private final AtomicInteger cacheTouch = new AtomicInteger(0);
+
+	private static final String GPX_METADATA_TABLE_NAME = "osm_gpx_data";
+	private static final String GPX_FILES_TABLE_NAME = "osm_gpx_files";
+
+	@GetMapping(path = {"/get-routes-list"}, produces = "application/json")
+	public ResponseEntity<String> getRoutes(@RequestParam String activity,
+	                                        @RequestParam(required = false) String year,
+	                                        @RequestParam String minlat,
+	                                        @RequestParam String maxlat,
+	                                        @RequestParam String minlon,
+	                                        @RequestParam String maxlon) {
+		if (!config.osmgpxInitialized()) {
+			return ResponseEntity.ok("OsmGpx datasource is not initialized");
+		}
+		cleanupCache();
+		String conditions = "";
+		conditions += " and m.maxlat >= " + minlat;
+		conditions += " and m.minlat <= " + maxlat;
+		conditions += " and m.maxlon >= " + minlon;
+		conditions += " and m.minlon <= " + maxlon;
```

Comment:
Please validate user-defined `minlat, maxlat, minlon, maxlon` by converting them to Float and then back to String. You should never allow invalidated user-defined variables in SQL-queries. Additionally, validate or use placeholders for `date` and `activities`.